scene/game/round_manager.gd

```diff
@@ -1,17 +1,27 @@
 class_name RoundManager
 extends Node
 
+signal round_goal_update
+
 @export var initialRoundGoal: ResourceProduction
 @onready var productionManager: ProductionManager = $"../ProductionManager"
-var roundCount = 1
-var currentRoundGoal: ResourceProduction
+@onready var gameManager: GameManager = $"../"
+var roundIndex = 0
+var roundGoals: Array[ResourceProduction] = []
 
 func _ready() -> void:
-	currentRoundGoal = initialRoundGoal
 	productionManager.round_goal_paid.connect(self._on_round_goal_paid)
+	gameManager.start_game.connect(self._on_game_start)
```

Comment:
I much prefer this way of connecting signal over the `obj.connect(signal_name, Callable)` way. Need to move my other signal code to this manner.